src/cosmology/compat/camb/flrw.py

```diff
@@ -0,0 +1,416 @@
+"""The Cosmology API compatability wrapper for CAMB."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+from cosmology.api import FLRWAPIConformantWrapper
+from cosmology.compat.camb import constants
+from cosmology.compat.camb.core import CAMBCosmology
+from typing_extensions import TypeAlias
+
+if TYPE_CHECKING:
+    from numpy import floating
+    from numpy.typing import NDArray
+
+    NDFloating: TypeAlias = NDArray[floating[Any]]
+
+
+__all__: list[str] = []
+
+
+_MPCGYR_KMS = np.array("9.7309928209912e35", dtype=np.float64)  # [Mpc / km]
+
+
+@dataclass(frozen=True)
+class CAMBFLRW(CAMBCosmology, FLRWAPIConformantWrapper):
+    """Cosmology API Protocol for FLRW-like cosmologies.
+
+    This is a protocol class that defines the standard API for FLRW-like
+    cosmologies. It is not intended to be instantiaed. Instead, it should be
+    used for ``isinstance`` checks or as an ABC for libraries that wish to
+    define a compatible cosmology class.
+    """
+
+    @property
+    def H0(self) -> NDFloating:
+        """Hubble constant at z=0 in km s-1 Mpc-1."""
+        return np.asarray(self._params.H0)
```

Comment:
IMO 2 things:
1. Consistency. Everything has the same attributes, like ``.shape``
2. Faster type promotion.
3. Non-scalars. Over at Astropy I've been slowly refactoring stuff so that we can do Cosmologies on a grid of parameters. So having 0d array is more consistent long term.

Reply:
Sounds good, point 3 in particular ...